presto-raptor/src/main/java/com/facebook/presto/raptor/storage/ShardRecoveryManager.java

```diff
@@ -169,7 +169,8 @@ private synchronized void enqueueMissingShards()
                 stats.incrementBackgroundShardRecovery();
                 Futures.addCallback(
                         shardQueue.submit(new MissingShard(shard.getShardUuid(), shard.getCompressedSize(), shard.getXxhash64(), false)),
-                        failureCallback(t -> log.warn(t, "Error recovering shard: %s", shard.getShardUuid())));
+                        failureCallback(t -> log.warn(t, "Error recovering shard: %s", shard.getShardUuid())),
```

Comment:
See if you can switch this to `addExceptionCallback()` from `MoreFutures`, then we can delete `failureCallback()`.